### DISCUSSION TITLE
Are there a way to use environment variables in the project file?

### DISCUSSION THREAD

**Discussion by devunt** [ariga/atlas#1573] [Q&A]
Are there a way to use the environment variables in the `atlas.hcl`?

```
env "dev" {
  src = "./schema.hcl"
  url = env("DATABASE_URL");
}
```

I would like to write something like above.
Or we can provide `var` from environment variables, something like this.

```
variable "url" {
  type = string
}

env "dev" {
  src = "./schema.hcl"
  url = var.url;
}
```

and run with `ATLAS_VAR_url=mysql://... atlas migrate diff`.
Packer [already does that](https://developer.hashicorp.com/packer/docs/templates/hcl_templates/variables#assigning-values-to-input-variables), so it seems reasonable to support the feature.

**Comment by andrew-womeldorf** [comment]
```
variable "url" {
  type = string
  default = getenv("DATABASE_URL")
}

env "dev" {
  url = var.url
}
